keep/common/management/commands/update_pidman_title.py

```diff
@@ -0,0 +1,247 @@
+    as any errors (exceptions) that may have occured.
+
+    Dry run is an option that only generates the report without applying
+    any changes. Please be sure that you know what you are doing, and
+    have the Fedora and Pidman environments set up correctly before
+    you proceed. You can use '-n' or '--dry-run' to enable the dry run
+    mode.
+
+    """
+
+    # dry-run option declaration
+    option_list = BaseCommand.option_list + (
+        make_option('--dry-run',
+            '-n',
+            action='store_true',
+            help='Dry run the command to get an output but with no changes applied'),
+    )
+
+    def handle(self, *args, **kwargs):
+        # disable info messages in the console
+        logging.getLogger('requests').setLevel(logging.CRITICAL)
+
+        # initialize a Pidman client object
+        self.pidman = self.get_pidman()
+
+        # initialize a Fedora repository object
+        self.repo = Repository()
+
+        # send greeting message with instructions
+        self.stdout.write(self.greetings)
+
+        # initialize dry-run flag to false
+        self.is_dry_run = False
+
+        # dry run notice
+        if kwargs.get('dry_run', True):
+            dry_run_notice = """
+            [DRY-RUN ACTIVATED] Currently you are running the script
+            in dry-run mode. It means that no changes will be applied
+            to any of the repositories except for that the script will
+            still generate statistics about objects in your repositories.
+
+            """
+            sys.stdout.write(dry_run_notice)
+            self.is_dry_run = True
+
+        # verify environments
+        environment_notice = """
+        Please confirm that you would like to proceed with the
+        following environments:
+        FEDORA: %s
+        PIDMAN: %s
+        """ % (settings.FEDORA_ROOT, settings.PIDMAN_DOMAIN)
+        self.stdout.write(environment_notice)
+
+        # confirm to proceed
+        confirmation = raw_input("Verify above information and type 'yes' to proceed: ")
+        if (confirmation != "yes"):
+            self.stdout.write("Answer is not 'yes'. Operation aborted.")
+            sys.exit(1)
+
+        # setup output directory paths
+        self.current_time_string = time.strftime("%Y%m%d-%H%M%S", time.localtime())
+        self.output_path = "%s/%s/%s/" % (os.getcwd(), "tmp", self.current_time_string)
+        self.error_path = "%s/errors/" % self.output_path
+
+        # create output directories
+        if not os.path.exists(self.output_path):
+            os.makedirs(self.output_path)
+        if not os.path.exists(self.error_path):
+            os.makedirs(self.error_path)
+
+        # start the object collection process
+        self.stdout.write("Started collecting objects from Fedora...")
+
+        # get objects of different content models
+        audio_objects = self.repo.get_objects_with_cmodel(AudioObject.AUDIO_CONTENT_MODEL)
```

Comment:
Haven't found a simple example; we're doing something along these lines in the eulindexer reindex script, but that's much more complicated than we need here.  What I think you want to do is basically what the server [get_objects_with_cmodel method](https://github.com/emory-libraries/eulfedora/blob/master/eulfedora/server.py#L270) does internally and just not create a list, because the risearch method returns a generator.   We can also get a count from the risearch, although there's not currently a method that exposes that for you directly.  (There's a count query, but you have to generate the query.)

It would be nice if these helper methods were available in eulfedora, but I'm not certain where they would belong.  Perhaps we could change `get_objects_with_cmodel` to return a generator instead of a list (since you can always convert a generator to a list), but we'd have to check if that causes any problems elsewhere.